subquery/src/mappings/nodl.ts

```diff
@@ -64,9 +64,15 @@ export async function handleERC20Transfer(event: TransferLog): Promise<void> {
       value
     );
 
-    from.balance =
+    const fromNewBalance =
       (from.balance || BigInt(0)) - (value > 0 ? value : BigInt(0));
-    to.balance = (to.balance || BigInt(0)) + value;
+    const toNewBalance = (to.balance || BigInt(0)) + value;
+
+    await handleLevel(fromNewBalance, from.balance || BigInt(0), timestamp);
+    await handleLevel(toNewBalance, to.balance || BigInt(0), timestamp);
+    
+    from.balance = fromNewBalance < 0 ? BigInt(0) : fromNewBalance;
```

Comment:
Again I'd rather get error on unexpected values. If the indexer is correctly synced starting from before the contract's start time, then fromNewBalance should never be < 0. Please add an error for fromNewBalance < 0 and then simplify this line as:
```suggestion
    from.balance = fromNewBalance;
```

Reply:
This is to prevent negatives, would like to use "Math.max" but is not compatible with BN. Added a comment

Reply:
I understand. I'm saying if it's negative through an error. If it's not go ahead. But do not assume a negative value as zero solely because it's unexpected.